### PR TITLE
feat(content): provide slug based on filename

### LINKED ISSUE #247: add slug based on filename to ContentFile data
### Which scope/s are relevant/related to the feature request?

content

### Information

The slug for the URL of a content page (markdown) can currently be explicitely added using a frontmatter  attribute or by extracting it form the `filename` which is part of the `ContentFile` interface. It would be neat it the slug can be provided by the content plugin based on the filename without removing manually the path and extension parts form the `filename` string.

**Suggestion**

```diff

export interface ContentFile< Attributes extends Record<string, any> = Record<string, any>> {
  filename: string;
+ slug: string;
  content?: string;
  attributes: Attributes;
}
```

**Example**

```diff
{
  attributes: { ... }
  filename: "/src/content/2023-foo-bar.md",
+ slug: "2023-foo-bar"
}
```

The slug should be URLEncoded.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by d-koppenhagen**
closes #247

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-angular-plugin
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [x] content

## What is the current behavior?

Issue Number: #247 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/content/src/lib/content-files-list-token.ts [hunk 1]
```diff
@@ -3,6 +3,11 @@ import { InjectionToken } from '@angular/core';
 import { ContentFile } from './content-file';
 import { getContentFilesList } from './get-content-files';
 
+function getSlug(filename: string) {
+  const parts = filename.match(/^(\\|\/)(.+(\\|\/))*(.+)\.(.+)$/);
```
Comment (brandonroberts) [inline packages/content/src/lib/content-files-list-token.ts:7]: Will this support nested files?

Such as

/src/content/posts/my-post.md

Some tests for the regex would be nice also

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: packages/content/src/lib/content-files-list-token.ts (hunk 1 above)
Comment (d-koppenhagen) [reply packages/content/src/lib/content-files-list-token.ts:7]: Yes it will. Also Windows / Unix Path names. But I agree: tests would be neat.

**Comment by brandonroberts** [comment]
@d-koppenhagen there's a build failure happening, will you take a look?

**Review by brandonroberts — APPROVED** [review]
LGTM

**Comment by brandonroberts** [comment]
Thanks! @allcontributors add @d-koppenhagen for code

**Pull request merged**